### PR TITLE
create evaluation pipeline

### PR DESCRIPTION
**Pull request by mmaecki**

### CODE REVIEW

**Comment by mmaecki** [comment]
https://github.com/kordc/flavor-seek/pull/13/commits/627dcec2841b9e97707414eaf2c429e25ac11fe4 - search size of 1000 gives pretty good approximation of the metrics, and still runs quite quickly

File: scripts/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    return mrr, hits_at_1, hits_at_5, hits_at_10
+
+
+if __name__ == "__main__":
+    # Enter your paths, and search size here
+    path_to_queries_embeddings = "../data/query_embeddings.parquet"
+    path_to_recipes_embeddings = "../data/recipe_embeddings.parquet"
+    size_of_search = 1000
+
+    queries = pq.read_table(path_to_queries_embeddings).to_pandas()
```
Comment (SebChw) [inline scripts/evaluate.py:60]: pd has `pd.read_parquet` function

Comment (mmaecki) [reply scripts/evaluate.py:60]: Good to know. I'll change that

File: scripts/evaluate.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import pyarrow.parquet as pq
+import numpy as np
+import faiss
+
+
+def calculate_metrics(query_embeddings, doc_embeddings, true_indices, k):
+    dimension = query_embeddings.shape[1]
+    print("Creating index...")
+    # The search is performed using a dot product. This is equivalent to cosine similarity if the vectors are normalized.
+    index = faiss.IndexFlatIP(dimension)
```
Comment (kordc) [inline scripts/evaluate.py:10]: It's just on the CPU. Will you add a version for faster search with GPU?

Comment (mmaecki) [reply scripts/evaluate.py:10]: Afaik it only depends on what library you have installed.
`faiss-cpu` or `fais-gpu`

Comment (kordc) [reply scripts/evaluate.py:10]: not really. You need to add something like:

```py
if faiss.get_num_gpus() > 0:
    self.index = faiss.index_cpu_to_all_gpus(index)
else:
    self.index = index
```

**Pull request merged**